### PR TITLE
Some fixes from bugs found during PR #2518

### PR DESCRIPTION
**Pull request by fgalan**
Related with fails in csub update logic and a bug in cache update logic when condtions.attrs is empty.

### CODE REVIEW

File: test/functionalTest/cases/2507_csub_metadata_field/metadata_filter_existing_sub.test.DISABLED [hunk 1]
```diff
@@ -1,259 +0,0 @@
-# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
```
Comment (fgalan) [inline test/functionalTest/cases/2507_csub_metadata_field/metadata_filter_existing_sub.test.DISABLED]: It seems that the github heuristic to detect that a file has been renamed didn't detect the .test.DISABLED -> .test in this PR...

NTC (just a remark)

File: src/lib/mongoBackend/mongoUpdateSubscription.cpp [hunk 2]
```diff
@@ -618,15 +626,14 @@ void updateInCache
...
-    subCacheItemRemove(subCacheP);
-  }
-
   if (mscInsert == 0)  // 0: Insertion was really made
   {
     subCacheUpdateStatisticsIncrement();
+    if (subCacheP != NULL)
+    {
+      LM_T(LmtSubCache, ("Calling subCacheItemRemove"));
+      subCacheItemRemove(subCacheP);
```
Comment (fgalan) [inline src/lib/mongoBackend/mongoUpdateSubscription.cpp]: Moving the remove operation inside the `if (mscInsert == 0)` is not directly related with the fix implemented in this bug. However, I think is an important "safety measure" as in the future other conditions may fail at mongoSubCacheItemInsert() and we don't want in that case not only not updating the subscription but removing it from the cache.

Comment (kzangeli) [inline src/lib/mongoBackend/mongoUpdateSubscription.cpp]: Sure. Seems like a good idea

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**